components/connectionmanager/src/lib.rs

```diff
@@ -279,6 +271,42 @@ impl ConnectionManager {
         }
     }
 
+    /// Queries `num_seeders_to_query` DNS seeders in parallel
+    async fn dns_seed_many(self: &Arc<Self>, num_seeders_to_query: usize) -> usize {
+        info!("Querying {} DNS seeders", num_seeders_to_query);
+        let shuffled_dns_seeders = self.dns_seeders.choose_multiple(&mut thread_rng(), num_seeders_to_query);
+        let jobs = shuffled_dns_seeders.map(|seeder| {
+            let cmgr = self.clone();
+            tokio::task::spawn_blocking(move || cmgr.dns_seed_single(seeder))
+        });
+        try_join_all(jobs).await.unwrap().into_iter().sum()
+    }
+
+    /// Query a single DNS seeder and add the obtained addresses to the address manager.
+    ///
+    /// DNS lookup is a blocking i/o operation so this function is assumed to be called
+    /// from a blocking execution context.
+    fn dns_seed_single(self: &Arc<Self>, seeder: &str) -> usize {
+        info!("Querying DNS seeder {}", seeder);
+        // Since the DNS lookup protocol doesn't come with a port, we must assume that the default port is used.
+        let addrs = match (seeder, self.default_port).to_socket_addrs() {
```

Comment:
what about async call her with: 
https://docs.rs/tokio/latest/tokio/net/fn.lookup_host.html https://docs.rs/tokio/latest/tokio/net/trait.ToSocketAddrs.html


maybe unrelated change, does it worth TODO, wdyt?

Reply:
Cool. I think we were looking for something like this. I'll give it a try now

Reply:
I guess it doesn't work the way I thought about

Reply:
You tried?

Reply:
In rust playground. I only got errors.

That's what we can use:
https://docs.rs/trust-dns-resolver/0.23.2/trust_dns_resolver/struct.Resolver.html#method.lookup_ip

Reply:
That's for another day